kibana/analytics-scripting/analytics_script.py

```diff
@@ -0,0 +1,261 @@
+"""
+This file is for analytics scripting, querying data from the dev-skevents-* index pattern
+and placing the result in the dev-analytics-* index pattern for further visualizations
+"""
+
+import argparse
+from datetime import datetime, timedelta
+from dateutil import parser
+from opensearchpy import OpenSearch, helpers
+
+ANALYTICS_INDEX_PATTERN = "dev-analytics"
+SK_INDEX_PATTERN = "dev-skevents-*"
+DEFAULT_NUM_COMPOSITE_BUCKETS = 10
+FIVE_MINS = timedelta(minutes=5)
+
+query_avg_response_time = {
+    "query": {
+        "bool": {
+            "must": [{"match_phrase": {"flowId": {}}}, {"range": {"tsEms": {}}}],
+            "filter": [{"match_all": {}}],
+        }
+    },
+    "size": 0,
+    "aggs": {
+        "my_buckets": {
+            "aggs": {
+                "min": {"min": {"field": "tsEms"}},
+                "max": {"max": {"field": "tsEms"}},
+                "sessionLength": {
+                    "bucket_script": {
+                        "buckets_path": {"startTime": "min", "endTime": "max"},
+                        "script": "params.endTime - params.startTime",
+                    }
+                },
+            },
+            "composite": {
+                "sources": [{"agg": {"terms": {"field": "interactionId.keyword"}}}]
+            },
+        },
+        "hits": {"top_hits": {"_source": ["companyId", "flowId"], "size": 1}},
+    },
+}
+
+
+def add_keys_if_not_exist(dictionary: dict, keys: list):
+    """
+    Adds keys, in a nested format, to the dictionary if the keys do not exist.
+    """
+    curr_dict = dictionary
+    for key in keys:
+        curr_dict.setdefault(key, {})
+        curr_dict = curr_dict[key]
```

Comment:
This one took me a minute -- I would highly recommend renaming this method to `add_nested_keys()`, as `add_keys_if_not_exist` subtly implies that my dict will remain flat. I also think you can simplify your loop here. Isn't the following functionally equivalent to what you have now?

```suggestion
def add_keys_if_not_exist(dictionary: dict, keys: list):
    """
    Adds keys, in a nested format, to the dictionary if the keys do not exist.
    """
    curr_dict = dictionary
    for key in keys:
        curr_dict = curr_dict.setdefault(key, {})
```